test/stat_tracker_test.rb

```diff
@@ -51,4 +51,137 @@ def test_it_can_get_highest_total_score
+  end
+
+  def test_it_can_filter_away_games_by_team
+    assert_equal 7, @stat_tracker.away_games_filtered_by_team("19").count
+  end
+
+  def test_it_can_group_home_games_by_season
+    assert_equal true, @stat_tracker.home_games_grouped_by_season("19").keys.include?("20142015")
+    assert_equal true,
+    @stat_tracker.home_games_grouped_by_season("19").keys.include?("20162017")
+    assert_equal false, @stat_tracker.home_games_grouped_by_season("19").keys.include?(:tie)
+  end
+
+  def test_it_can_group_away_games_by_season
+    assert_equal true, @stat_tracker.away_games_grouped_by_season("19").keys.include?("20142015")
+    assert_equal true,
+    @stat_tracker.away_games_grouped_by_season("19").keys.include?("20122013")
+    assert_equal false, @stat_tracker.away_games_grouped_by_season("19").keys.include?(:home_win)
+  end
+
+  def test_it_can_get_number_of_home_wins_in_season
+    assert_equal true, @stat_tracker.season_home_wins("19").values.include?(2.0)
+  end
+
+  def test_it_can_get_number_of_away_wins_in_season
+    assert_equal true, @stat_tracker.season_away_wins("19").values.include?(1.0)
+  end
+
+  def test_it_can_get_total_wins_in_a_season
+    assert_equal true, @stat_tracker.win_count_by_season("19").values.include?(-1.0)
+  end
+
+  def test_it_can_get_best_season
+    assert_equal "20162017", @stat_tracker.best_season("19")
+  end
+
+  def test_it_can_get_number_of_home_losses_in_season
+    assert_equal true, @stat_tracker.season_home_losses("19").values.include?(-1.0)
+  end
+
+  def test_it_can_get_number_of_away_losses_in_season
+    assert_equal true, @stat_tracker.season_away_losses("19").values.include?(-1.0)
+  end
+
+  def test_it_can_get_total_losses_in_a_season
+    assert_equal true, @stat_tracker.loss_count_by_season("19").values.include?(1.0)
+  end
+
+  def test_it_can_get_worst_season
+    assert_equal "20122013", @stat_tracker.worst_season("19")
+  end
+
+  def test_it_can_get_all_games_played_by_a_team
+    assert_equal 12, @stat_tracker.combine_all_games_played("19").count
+  end
+
+  def test_it_can_total_wins_or_ties_for_a_team
+    assert_equal 0.5, @stat_tracker.find_total_wins_or_ties("19")
+  end
+
+  def test_it_can_get_average_win_percentage
+    assert_equal Float, @stat_tracker.average_win_percentage("19").class
+  end
+
+  def test_it_can_get_most_home_goals_scored
+    assert_equal 3, @stat_tracker.most_home_goals_scored("19")
+  end
+
+  def test_it_can_get_most_away_goals_scored
+    assert_equal 4, @stat_tracker.most_away_goals_scored("19")
+  end
+
+  def test_it_can_get_most_goals_scored
+    assert_equal 4, @stat_tracker.most_goals_scored("19")
+    assert_equal 3, @stat_tracker.most_goals_scored("30")
+    assert_equal 4, @stat_tracker.most_goals_scored("26")
+  end
+
+  def test_it_can_get_fewest_home_goals_scored
+    assert_equal 3, @stat_tracker.most_home_goals_scored("19")
+  end
+
+  def test_it_can_get_fewest_away_goals_scored
```

Comment:
I'm not exactly sure where, but I have a feeling some of these could make use of mocks and/or stubs to make tests run faster. The tests I wrote I'm not very proud of, as they don't seem very robust. Maybe a stub could be used to stub the output from a helper method or two, which would then be used by the final method required by Iteration 2?